historical_surveillance/views.py

```diff
@@ -0,0 +1,328 @@
+from django.contrib import messages
+from django.shortcuts import render, redirect
+from django.db.models import Count
+from .forms import AggregateEnrollmentForms, DistrictForms, SchoolForms, EnrollmentForms
+from .models import AggregateEnrollment, School, District, Enrollment
+import pandas as pd
+from .utils import get_plot, get_pairs
+
+
+# This is the view for the home page of this app
+def index(request):
+    return render(request, 'surv_home.html', {})
+
+
+# This function controls the district creation form view and process form data for save into the database
+def district(request):
+    error_message = None
+    form = DistrictForms()
+    data = District.objects.all()
```

Comment:
This line fetches all rows from the DB. And if this is being used as a post request that data isn't even being used.

I'd consider just moving this query to the end so we only query the data if its needed

```
return render(request, 'district.html', {'form': form, 'district_created': District.objects.all()})
```